add automation of the RGP proband tab

Rinchan20 opened the pull request:
is slightly flaky when checking that textarea notes are preserved after revisiting the page

is part 3 of breaking down this large PR: https://github.com/broadinstitute/ddp-angular/pull/2048

Rinchan20 committed:
add automation of the proband tab

is slightly flaky when checking that textarea notes are preserved after revisiting the page

Rinchan20 commented:
dev probably needs a slight tweak in order for the test to pass - doesn't seem to be housekeeping related - the tweak needed is that the RGP dynamic form -> Contact Info -> State is a textfield in dev env but is a dropdown in test env. It also appears to be a textfield in staging.  So it just needs consistency across the 3 non-prod envs. I can just change it so that it looks for a textfield instead of a dropdown though. Passes in test env though.

aweng98 commented:
I ran all tests against Dev on CircleCI. It had one test failure which I think it's related to your change. Can you take a look see to it can be fixed? 
https://app.circleci.com/pipelines/github/broadinstitute/ddp-angular/25247/workflows/2d615392-ce1c-4a0d-a23b-77f5d79655a8

Rinchan20 committed:
tweak to waiting for response

Rinchan20 commented:
if it's the failure for finding the proband tab - that one would be because of housekeeping. Once a participant enrolls in RGP, there should be a proband tab created in RGP but sometimes in dev (and recently test - at least on Friday, I think?) it will not appear without running DSM RGP export - I think it'll be fixed once this ticket is done: https://broadworkbench.atlassian.net/browse/PEPPER-692

aweng98 commented on playwright-e2e/utils/date-utils.ts:
```diff
@@ -48,3 +48,18 @@ export function offsetDaysFromToday(number = 1, opts: { isAdd?: boolean } = {}):
   return today;
 }
+
+export const calculateBirthDate = (month: string, day: string, year: string): number => {
```
This func is nice! 
How about rename to `calculateAge`? I think date is not accurate description for what the func do.

aweng98 commented on playwright-e2e/tests/rgp/dsm-family-enrollment.spec.ts:
```diff
@@ -78,4 +80,432 @@ test.describe.serial('DSM Family Enrollment Handling', () => {
+    await participantListPage.waitForReady();
+    const participantListTable = new ParticipantListTable(page);
+    await participantListPage.filterListByParticipantGUID(user.patient.participantGuid);
+    await participantListTable.openParticipantPageAt(0);
```
Need a check to assert one row exists after filter search before open participant page.

aweng98 commented on playwright-e2e/dsm/pages/participant-page/rgp/family-member-tab.ts:
```diff
@@ -0,0 +1,638 @@
+    }
+
+    public getPrimaryPhoneNumber(): Locator {
+        return this.page.locator("//td[contains(text(), 'Phone')]/following-sibling::td//div//input[@data-placeholder='Phone (Primary)']");
```
Not a fan of long xpath. It is brittle. We can improve it at a later time.

aweng98 commented on playwright-e2e/dsm/pages/participant-page/rgp/family-member-tab.ts:
```diff
@@ -0,0 +1,638 @@
+        this._relationToProband = relationToProband;
+    }
+
+    public set relationshipID(relationshipID : string) {
```
Why do we need these `set` functions?

Rinchan20 replied:
plan is to use them at around the next PR when adding family members - but since they're not relevant for this PR I think I'll just take them out and add them back in only if they're actually being used - thinking about it more, that'll help with not causing confusion down the line when seeing a method available that's not clearly being used

aweng98 reviewed (commented):
There is a problem with test which explains why it doesn't run on CircleCI and localhost. Failure root cause is not related to Housekeeping.
The `user` object inside the `saveParticipantGuid(participantGuid)`  function is not "known" to the test. `user` objects in the test and that func are two different objects. Let me know if you have any questions about this.

Rinchan20 committed:
Merge branch 'develop' into rgp-fill-out-proband-tab

Rinchan20 committed:
implemented feedback

works well for new pts - is iffy on existing pts since it seems to timeout when checking for a patch request

Rinchan20 committed:
taking out part that does nothing for flakiness

Rinchan20 committed:
implemented feedback

Rinchan20 committed:
some feedback and changes

Rinchan20 committed:
implemented feedback

Rinchan20 committed:
Merge branch 'develop' into rgp-fill-out-proband-tab

Rinchan20 committed:
hopefully fixed flakiness of text input into textarea

basically done by adding an additional check to make sure the text is there before proceeding

Rinchan20 committed:
forgot to get rid of method

Rinchan20 committed:
Merge branch 'develop' into rgp-fill-out-proband-tab

Rinchan20 committed:
Update dsm-family-enrollment.spec.ts

skipping until housekeeping stuff is fixed so that tests will pass consistently
